### PR TITLE
Added commands history for quick execution of old commands

### PR DESCRIPTION
**Pull request by VlachJosef**
For displaying of commands history differed functions (reads) are being tested in
this order:

1. ivy (ivy-read)
2. ido (ido-completing-read)
3. default (completing-read)

First function present is used.

### CODE REVIEW

File: sbt-mode-hydra.el [hunk 1]
```diff
@@ -339,21 +346,54 @@ x - clean        - reset substring (-- -z) to empty string
           (t
            (message "Do not know how to run project %s." project)))))
 
+(defun sbt-hydra:previous-command-play-run (play-project)
+  (string-match (format "%s/run" play-project) (car sbt-hydra:hydra-previous-commands)))
+
+(defun sbt-hydra:read-from-history ()
+  (cond ((fboundp 'ivy-read)
+         (ivy-read "History: " sbt-hydra:hydra-previous-commands))
```
Comment (yyadavalli) [inline sbt-mode-hydra.el]: If `ivy-mode` is enabled, `completing-read` automatically uses ivy. I don't think this is needed.

Comment (VlachJosef) [inline sbt-mode-hydra.el]: hm, I don't have `ivy-mode` enabled but I want to use it for sbt history if it is installed.

Comment (yyadavalli) [inline sbt-mode-hydra.el]: Hmm, in that case isn't it better to provide a variable/customization that enables ivy.

Comment (VlachJosef) [inline sbt-mode-hydra.el]: Yes, that would be right way to do it... to let user choose what to use rather then this custom fallback mechanism.

**Pull request merged**